src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,463 @@
+
+        // constructors
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServerVersion"/> class.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <param name="patch">The patch version.</param>
+        public ServerVersion(int major, int minor, int patch)
+            : this(major, minor, patch, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServerVersion"/> class.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <param name="patch">The patch version.</param>
+        /// <param name="releaseCandidate">The release candidate version.</param>
+        public ServerVersion(int major, int minor, int patch, int? releaseCandidate)
+            : this(major, minor, patch, releaseCandidate, null, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServerVersion"/> class.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <param name="patch">The patch version.</param>
+        /// <param name="internalBuild">The internal build version.</param>
+        /// <param name="commitHash">The internal build commit hash.</param>
+        public ServerVersion(int major, int minor, int patch, int? internalBuild, string commitHash)
+            : this(major, minor, patch, null, internalBuild, commitHash)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServerVersion"/> class.
+        /// </summary>
+        /// <param name="major">The major version.</param>
+        /// <param name="minor">The minor version.</param>
+        /// <param name="patch">The patch version.</param>
+        /// <param name="releaseCandidate">The release candidate version.</param>
+        /// <param name="internalBuild">The internal build version.</param>
+        /// <param name="commitHash">The internal build commit hash.</param>
+        public ServerVersion(int major, int minor, int patch, int? releaseCandidate, int? internalBuild, string commitHash)
+        {
+            _major = Ensure.IsGreaterThanOrEqualToZero(major, nameof(major));
+            _minor = Ensure.IsGreaterThanOrEqualToZero(minor, nameof(minor));
+            _patch = Ensure.IsGreaterThanOrEqualToZero(patch, nameof(patch));
+            _releaseCandidate = releaseCandidate; // can be null
+            _internalBuild = internalBuild; // can be null
+            _commitHash = commitHash; // can be null
+        }
+
+        // properties
+        /// <summary>
+        /// Gets the internal build commit hash.
+        /// </summary>
+        /// <value>
```

Comment:
as far as I know, we don't set `<value>` comments any more

Reply:
We are not consistent. I don't think `<value>` is helpful but I think GhostDoc generates them.